internal/transport/http2_client.go

```diff
@@ -1695,8 +1704,16 @@ func (t *http2Client) reader(errCh chan<- error) {
 	}
 }
 
+func (t *http2Client) reader(errCh chan<- error) {
+	err := t.readerUtil(errCh)
+	if err != nil {
+		t.Close(err)
+	}
+}
+
 // keepalive running in a separate goroutine makes sure the connection is alive by sending pings.
-func (t *http2Client) keepalive() {
+func (t *http2Client) keepaliveUtil() error {
```

Comment:
Similar comment as above re: this pattern.

Reply:
Done